src/services/health-check/health-check.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Yoast\WP\SEO\Services\Health_Check;
+
+/**
+ * Abstract class for all health checks. Provides a uniform interface for the Health_Check_Integration.
+ */
+abstract class Health_Check {
+
+	/**
+	 * The prefix to add to the test identifier. Used to differentiate between Yoast's health checks, and other health checks.
+	 */
+	const TEST_IDENTIFIER_PREFIX = 'yoast-';
+
+	/**
+	 * The object that runs the actual health check.
+	 *
+	 * @var Runner_Interface
+	 */
+	private $runner;
+
+	/**
+	 * The health check implementation sets the runner so this class can start a health check.
+	 *
+	 * @param  Runner_Interface $runner The health check runner.
+	 * @return void
+	 */
+	protected function set_runner( $runner ) {
+		$this->runner = $runner;
+	}
+
+	/**
+	 * Returns the identifier of health check implementation. WordPress needs this to manage the health check (https://developer.wordpress.org/reference/hooks/site_status_tests/).
+	 *
+	 * @return string
+	 */
+	public function get_test_identifier() {
+		$label                = $this->get_test_label();
+		$lowercase            = strtolower( $label );
+		$whitespace_as_dashes = str_replace( ' ', '-', $lowercase );
+		$with_prefix          = self::TEST_IDENTIFIER_PREFIX . $whitespace_as_dashes;
+		return $with_prefix;
+	}
+
+	/**
+	 * Returns the name of health check implementation that the user can see. WordPress needs this to manage the health check (https://developer.wordpress.org/reference/hooks/site_status_tests/).
+	 *
+	 * @return string
+	 */
+	abstract public function get_test_label();
+
+	/**
+	 * Runs the health check, and returns its result in the format that WordPress requires to show the results to the user (https://developer.wordpress.org/reference/hooks/site_status_test_result/).
+	 *
+	 * @return string[]
+	 */
+	public function run_and_get_result() {
+		$this->runner->run();
```

Comment:
Yes I have. That would mean that this class would have to know about the implementation of a runner, because different runners would return different results.

I could defer this method to every subclass of `Health_Check` but I'd like to keep those as small controller classes and DRY. Also, if I then want to retain the separation between the runner and its reports, I'd have to introduce another class to hold the health check's result. I could make it an anonymous object/array, but I'm really trying to avoid those.

Having a default (unused) state for a runner isn't perfect, but since it is always run first, and this class doesn't have to change, I don't think it's it's likely to cause issues.